app/jobs/data_clear_jobs/clear_user_data.rb

```diff
@@ -17,8 +17,9 @@ def perform(intervention_id)
 
   private
 
-  def delete_quest_users_without_any_intervention!
+  def delete_temporary_users_without_any_intervention!
     User.left_joins(:user_interventions).where(roles: ['guest'], user_interventions: { id: nil }).destroy_all
+    User.left_joins(:user_interventions).where(roles: ['predefined_participant'], user_interventions: { id: nil }).destroy_all
```

Comment:
I think we can combine these two requests, you can look at https://stackoverflow.com/questions/69536519/how-to-get-all-elements-where-one-string-array-contains-any-of-the-elements-in-a for inspiration